contrib/babelfishpg_common/src/varbinary.c

```diff
@@ -636,20 +641,26 @@ varcharvarbinary(PG_FUNCTION_ARGS)
 						"varbinary is not allowed. Use the CONVERT function "
 						"to run this query.")));
 
-	/* If typmod is -1 (or invalid), use the actual length */
+	collInfo = lookup_collation_table(get_server_collation_oid_internal(false));
```

Comment:
We can have different collation on this source varchar column

Reply:
Collation does not affect the encoding unless we add the UTF8 option in collation ex: Japanese_Bushu_Kakusu_100_CS_AS_KS_WS_UTF8
[Reference Link](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/relational-databases/collations/collation-and-unicode-support.md#examples-of-collation-options)

Reply:
can u add tests for this, where collation of source varchar column is different than server collation.